src/basic/fd-util.c

```diff
@@ -48,7 +48,7 @@ int close_nointr(int fd) {
          * http://utcc.utoronto.ca/~cks/space/blog/unix/CloseEINTR
          * https://sites.google.com/site/michaelsafyan/software-engineering/checkforeintrwheninvokingclosethinkagain
          */
-        if (errno == EINTR)
+        if (errno == EINTR || errno == EINPROGRESS)
```

Comment:
Please use `IN_SET(errno, EINTR, EINPROGRESS)`.
Also add comments about EINPROGRESS in the code.